python/GafferImageTest/ImageWriterTest.py

```diff
@@ -1495,25 +1496,105 @@ def testReproduceWeirdPartNames( self ):
 		writer = GafferImage.ImageWriter()
 		writer["in"].setInput( collect["out"] )
 		writer["fileName"].setValue( os.path.join( self.temporaryDirectory(), "test.exr" ) )
-		writer["namingConvention"]["partName"].setValue( '${naming:layerWithDefault}.main' )
-		writer["namingConvention"]["channelName"].setValue( '${naming:layer}.${naming:channel}' )
+		writer["layout"]["partName"].setValue( '${imageWriter:standardPartName}.main' )
+		writer["layout"]["channelName"].setValue( '${imageWriter:layerName}.${imageWriter:baseName}' )
 		writer["expression"] = Gaffer.Expression()
 		writer["expression"].setExpression(
-			'c = context["naming:channel"]; l = context["naming:layer"]; parent["namingConvention"]["partName"] = "part%i" % ( "RGBA".find(c) + 2 * ( l == "layer" ) )'
+			'c = context["imageWriter:baseName"]; l = context["imageWriter:layerName"]; parent["layout"]["partName"] = "part%i" % ( "RGBA".find(c) + 2 * ( l == "layer" ) )'
 		)
 		writer["task"].execute()
 
 		refReader = GafferImage.ImageReader()
 		refReader["fileName"].setValue( "${GAFFER_ROOT}/python/GafferImageTest/images/weirdPartNames.exr" )
 
 		rereader = GafferImage.ImageReader()
-		rereader["channelNaming"].setInput( refReader["channelNaming"] )
+		rereader["channelInterpretation"].setInput( refReader["channelInterpretation"] )
 		rereader["fileName"].setInput( writer["fileName"] )
 
-		refReader["channelNaming"].setValue( GafferImage.OpenImageIOReader.ChannelNaming.Default )
+		refReader["channelInterpretation"].setValue( GafferImage.ImageReader.ChannelInterpretation.Default )
 		self.assertImagesEqual( rereader["out"], refReader["out"], ignoreMetadata = True )
-		refReader["channelNaming"].setValue( GafferImage.OpenImageIOReader.ChannelNaming.Legacy )
+		refReader["channelInterpretation"].setValue( GafferImage.ImageReader.ChannelInterpretation.Legacy )
 		self.assertImagesEqual( rereader["out"], refReader["out"], ignoreMetadata = True )
 
+	def testWithChannelTestImage( self ):
+
+		# Helper function that extracts the part of the header we care about from exrheader
+		def usefulHeader( path ):
+			r = subprocess.check_output( ["exrheader", path ] ).splitlines()[2:]
+
+			r = [ i.decode("utf-8") for i in r ]
```

Comment:
I think we can pass `universal_newlines=True` to `check_output` and it will do this for us, but being a bit smarter about non-utf8 locales. It's a weird parameter, but is also available as the more sensibly named `text` in Python 3. So when we can finally ditch Python 3 it'll be easy to do a search-replace.